cmd/manager/main.go

```diff
@@ -139,8 +139,9 @@ func main() {
 		log.Info("Could not create metrics Service", "error", err.Error())
 	}
 	// Create ServiceMonitor pointing to the metrics service.
-	serviceMonitor := metrics.GenerateServiceMonitor(service)
-	if err := mgr.GetClient().Create(context.TODO(), serviceMonitor); err != nil {
+	services := []*v1.Service{service}
+	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+	if err != nil {
```

Comment:
Can you follow my if/err pattern? I think that should work just as well here.

Reply:
updated